src/rescorer/score_collector.cpp

```diff
@@ -91,10 +91,10 @@ void ScoreCollectorNBest::Write(long id,
                "Entry {} < {} already read but not in buffer",
                id,
                lastRead_);
-      std::string line;
-      while(lastRead_ < id && utils::GetLine((std::istream&)*file_, line)) {
+      std::string line1;
```

Comment:
(I presume this is due to shadowing. Consider to simply not dedeclare this variable, and just use the previous `line`, as long as the code remains correct. I did a similar change elsewhere.)